beacon_node/eth2_libp2p/src/rpc/codec/ssz_snappy.rs

```diff
@@ -110,17 +111,24 @@ impl<TSpec: EthSpec> Decoder for SSZSnappyInboundCodec<TSpec> {
 
         let length = self.len.expect("length should be Some");
 
-        // Should not attempt to decode rpc chunks with length > max_packet_size
-        if length > self.max_packet_size {
+        // Should not attempt to decode rpc chunks with `length > max_packet_size` or not within bounds of
+        // packet size for ssz container corresponding to `self.protocol`.
+        let ssz_limits = self.protocol.rpc_request_limits();
+        if length > self.max_packet_size || length > ssz_limits.1 || length < ssz_limits.0 {
```

Comment:
When `ssz_limits.1` is 0, the length shouldn't have been read in the first place. I.e. if a metadata request (which has no contents) is received, it shouldn't contain any bytes.

Reply:
@protolambda metadata requests do not reach the decoder. They return early here
https://github.com/sigp/lighthouse/blob/56ffe91f9093605737ca1c676ea3493a75b2cfe6/beacon_node/eth2_libp2p/src/rpc/protocol.rs#L224-L226